jump/src/installer.rs

```diff
@@ -149,19 +202,43 @@ pub(crate) fn install(files: &[FileEntry], payload: &[u8]) -> Result<(), String>
                 let bytes = &payload[location..(location + tote_file.size)];
                 unpack(
                     tote_file.file_type,
-                    Cursor::new(bytes),
+                    || Ok((Cursor::new(bytes), ())),
                     tote_file.hash.as_str(),
                     &scie_tote_path,
                 )?;
                 for (file, dst) in entries {
-                    let src_path = scie_tote_path.join(&file.name);
-                    let src = std::fs::File::open(&src_path).map_err(|e| {
-                        format!(
-                            "Failed to open {file:?} at {src} from the unpacked scie-tote: {e}",
-                            src = src_path.display()
-                        )
-                    })?;
-                    unpack(file.file_type, &src, file.hash.as_str(), dst)?;
+                    let scie_tote_src = || {
+                        let src_path = scie_tote_path.join(&file.name);
+                        let file = std::fs::File::open(&src_path).map_err(|e| {
+                            format!(
+                                "Failed to open {file:?} at {src} from the unpacked scie-tote: {e}",
+                                src = src_path.display()
+                            )
+                        })?;
+                        let permissions = file
+                            .metadata()
+                            .map_err(|e| {
+                                format!(
+                                    "Failed to read permissions of {src_path}: {e}",
+                                    src_path = src_path.display()
+                                )
+                            })?
+                            .permissions();
+                        Ok((file, permissions))
+                    };
+                    if let Some(permissions) =
+                        unpack(file.file_type, scie_tote_src, file.hash.as_str(), dst)?
+                    {
+                        // We let archives handle their internally stored permission bits.
```

Comment:
This was a pre-existing bug the load example exposed. In that case, the perms of the get.sh script were not preserved from the scie-tote. A similar problem exists for non-scie-tote scies with blobs that should be executable. That is tracked in #34.